### ISSUE TITLE
Add mechanism to change the height of the entire room by x units.

### ISSUE THREAD

**Issue by skittles1** [OpenMeridian105/GZDoomBuilder_M59#9]
Not sure if it exists already, but a way to raise/lower every sector (floor and ceiling) by x units so the height of each sector stays the same would be useful.

**Comment by achrapko** [comment]
We discovered how to do this the other day - selecting any sectors you want to adjust and entering "++n" or "--n" in both the floor and ceiling to raise or lower the entire sector(s) by 'n' units.

This is not possible with slopes though, so it would be nice to have there. Though editing slopes is a huge pain anyways because of the tendency for the 3 vertices to be cleared from the dropdown selection

**Comment by skittles1** [comment]
Yeah the slope handling isn't the best, I had to replicate the way slopes were done in old room editor (i.e. not in the best way), hoping eventually to move both editors to the normal GZDoomBuilder way of doing slopes.

It should be renumbering slope vertexes correctly when you copy them, even between files - the caveat here is that the vertexes need to be part of the sloped sector for this to happen, and unfortunately in a lot of existing maps the slope vertexes are picked from other random sectors, so they have to be manually fixed. The new editor only allows picking from vertexes that are part of the slope so this can't happen in the future (old editor of course will still let you do it).

I [added some code now](17) to check for broken vertex references and attempt to repair them, it looks like it'll get about half the broken ones but require the rest to be fixed manually. Basically there's two types of broken references:
- Referencing wrong vertex (outside sector) but using the height of a vertex inside the sector.
- Referencing wrong vertex, and using the height at that vertex or a random height.

The first type can be fixed by switching the reference to an in-sector vertex with matching height. The second can't be fixed programmatically since a complex slope might have multiple choices for a new height. I suspect there's rare cases where fixing the first type might lead to an error (although I can't find any in testing) so it's probably a good idea to check any modified slopes.

I also added the [slope height increment/decrement](18) so using ++/-- on the floor/ceiling height will also modify any floor or ceiling slopes. New editor should be out in about an hour.
